### ISSUE TITLE
Bricked my gameshark pro

### ISSUE THREAD

**Issue by CanadianJeff** [parasyte/n64rd#3]
my gameshark pro was running version 3.30 I used the upgrade -u and flashed version 2 by mistake and now I just get...

```
Using port /dev/parport0...
Uploading `gs.n64`...
gspro.c:426, gs_enter() ERROR: gspro.c:136, _gs_exch_4(): Communications link timed out.
upgrade(): gs_enter() failed
n64rd v0.2
By Parasyte (parasyte@kodewerx.org)
Website: http://www.kodewerx.org/
Build date: Feb 20 2016, 00:16:52
```

**Comment by parasyte** [comment]
There is a way to refurbish bricked GameSharks by desoldering the flash ROMs and flashing them with an inexpensive programmer like [this one](http://www.ebay.com/itm/USB-MiniPro-TL866CS-Universal-BIOS-Programmer-EEPROM-FLASH-8051-AVR-GAL-PIC-SPI-/221920274094?hash=item33ab7aa6ae:g:imwAAOSwI-BWKNCL). (I use the [Genius G540 programmer](http://www.ebay.com/itm/Genius-G540-USB-Universal-Bios-GAL-Programmer-EPROM-FLASH-51-AVR-PIC-MCU-SPI-/321347814855?hash=item4ad1d285c7:g:9EMAAOxyfSpSTjIO))

When I do these repairs, I like to socket the PLCC32s, which makes it a piece of cake to repair them again later.

Here's a series of photos showing the process:

I used a heat gun to melt the solder, and a vacuum pen to remove the chips after they had loosened. After cleaning the solder from the pads with a little flux and solder braid:

![img_4645](https://cloud.githubusercontent.com/assets/456942/13200806/fc9de7ca-d80c-11e5-8492-fac999af94c4.JPG)

Add a line of solder paste to the cleaned pads:

![img_4647](https://cloud.githubusercontent.com/assets/456942/13200811/41bdfa48-d80d-11e5-9712-07be632f19e9.JPG)

Place the socket. (I had to cut away the plastic slightly to make room for the small capacitor at the bottom.) Pin 1 points down:

![img_4650](https://cloud.githubusercontent.com/assets/456942/13200814/5495b390-d80d-11e5-80b3-bacf8a985e92.JPG)

Apply hot air with the heat gun until the solder paste melts. I let the heat gun continue for a few seconds after melting to help eliminate solder bridges:

![img_4652](https://cloud.githubusercontent.com/assets/456942/13200822/970aa1f4-d80d-11e5-9fc8-d748c21fce71.JPG)

Repeat the process on the second chip. Do your final bridge checks with a multimeter, and you're all set!

![img_4654](https://cloud.githubusercontent.com/assets/456942/13200823/bb0ee902-d80d-11e5-9a23-ec1037c512cb.JPG)

As I recall, the chip closest to the GAL (the lower socket in the photos) is for high bytes (big-endian!), and the chip farthest from the GAL is for the low bytes. Here's a simple Python script that will split the firmware file into `filename.hi` and `filename.lo` files to be flashed individually: https://gist.github.com/parasyte/b9e2b7cca58eebfa2ec4 (And another script that can combine them, for completeness's sake. Useful for dumping ROMs from v2.x GameSharks.)

And here's the (currently to our knowledge) complete set of unencrypted GS Pro firmwares: http://www.kodewerx.org/downloads/n64-gspro-3.x-20160220.tar.bz2

**Comment by CanadianJeff** [comment]
piggyback method wont work? sorry I have zero soldering skills

I also did manage to dump my gameshark v3.30 before I flashed 2.0 by mistake

**Comment by parasyte** [comment]
Piggyback flashing was discussed in #n64dev (EFnet) yesterday. I never got it to work right, and I suspect it's due to bus conflicts while programming the EEPROMs; Two EEPROMs are tied to the same data pins on the PI bus when stacked, but they are not guaranteed to have identical timing (in fact, it's pretty much impossible that they will have identical timing).

That means one chip always enters programming mode first, one or the other always completes the write first, and either of them will report "write completed" status first. The last case puts two conflicting values on the data bus simultaneously. At best, one of the chips will hold the bus in the "still writing" status, but it's really undefined behavior, and the running firmware can potentially see either value, or a strange mixture of the two.

Anyway, you're welcome to try it, at your own risk. Also there are a few of us who will do the soldering work for you, but it may end up more expensive than just buying a new GameShark; shipping costs, materials, installation time.

**Comment by CanadianJeff** [comment]
could you hook up some jumper wires and flash it without removing the chip
would DATEL even release the tech sheet for the gameshark pro
to find out which pins on the chip are used to flash it
because me and my friends are too scared to desolder the chip
and it might just wind up being cost effective to buy a 2nd one and try the piggy back flash

**Comment by parasyte** [comment]
The flash write commands can only be sent over the PI bus (the cartridge edge connector). I know marshallh has been working on a little project that allows connecting N64 carts to a computer over USB. That would allow fixing bricked GameSharks. It's still in early phases, though.

It will be difficult to program the GameShark through the PI bus unless you have a programmer that uses the PI bus protocol. The other access point for programming is probing the chips directly, which is what we do by removing them from the PCB. They can safely be programmed in place, but it will be a challenge to jumper all of the pins without soldering or a bunch of test hook probes to grab the legs.

**Comment by sharkbyte64** [comment]
I read this topic a few weeks because I am having the same issue. I have about 9 bricked gamesharks that need repaired. I bought the G540 programmer you use and desoldered the chips on the gameshark. My soldering skills are poor since I have only done it once before. I think I ruined the chip in the process. The chip barely fits in the adapter that the programmer uses. I would like to know if there has been any progress from marshallh with reading n64 carts to a computer over USB. I really want to avoid desoldering and soldering as when I desoldered the chips the pads went all over the place and are now lost. I would be open to having someone repair them for me if they are reputable and a great solderer and send it back with a low cost. I rather do it myself but I want to avoid soldering at any cost, if possible. Do you know if its possible now?

**Comment by parasyte** [comment]
You might want to buy an inexpensive rework station; they are about $50 on eBay. The heat only needs to be like 300°F, and you only need to heat the solder for about 10 seconds. Try to avoid heating the chip. You'll also want to clean up the solder from the chip pins with an iron and solder braid. If you need replacement chips, we found that [SST39LF040](https://www.digikey.com/product-detail/en/microchip-technology/SST39LF040-55-4C-NHE/SST39LF040-55-4C-NHE-ND/2486469) are perfect drop-in replacements, and they only cost about $1.50 each.

[Here are the sockets](https://www.digikey.com/product-detail/en/assmann-wsw-components/A-CCS32-Z-SM-R/AE10070-ND/821824) ($1.46 ea.) and here's the [solder paste](https://www.digikey.com/product-detail/en/chip-quik-inc/SMDLTLFP/SMDLTLFP-ND/2682721) ($16).

The chips should fit very tightly into the sockets (and the programmer adapter). You should also have a [PLCC extractor tool](https://www.digikey.com/product-detail/en/jonard-tools/EX-6/K374-ND/135076) to remove the chips safely.

IMHO, flashing the chips out-of-band is the best way to repair bricked GameSharks (or any hardware, for that matter).

If all else fails, I can do these repairs for a modest fee (parts and labor). If interested, contact me at jay@kodewerx.org

**Comment by sharkbyte64** [comment]
Thanks for the links. The only tool I don't think I need is the PLCC extractor as it came with the set of that programmer. I'm not sure which corner I'm supposed to use it from as it comes loose very easily when I tried extracting the chip. Its hard to grab ahold of. Would those other chips you linked allow extra data for the codelist so I could have more codes? Will they really work with my programmer and be possible to reflash the gameshark rom so that the n64 reads it? I found a hot air gun on ebay but I'd like to know if you'd recommend it. Also I'd like to know what temperature you use to desolder as well as solder back on using the heat gun. Is it is celsius or fahrenheit? I am a huge fan of your work you've done on the n64 and have been meaning to start an account on kodewerx but just haven't done it. I found out about 6 years ago that the non shiny gamesharks have chips removed to use the parallel port. I remember years ago when I was trying to hack codes from the parallel port I couldn't get it to work and thought it was because I had the wrong cable or some settings weren't set right with my computer. It turns out I was wrong and bricked the gameshark trying a piggyback reflash later on with a working gameshark. Then after I bricked them I read that topic. I wish i read it before trying that. The heat gun I'm looking at is here. 

http://www.ebay.com/itm/858D-Rework-Station-Hot-Air-Gun-Kit-SMD-Iron-Soldering-Solder-Holder-700W-/272428095142?hash=item3f6dfad6a6:g:Do0AAOSwHsRYDbVE

I looked all over in my town and couldn't find any place that sells them so I guess online is my only option.

Also why do I need to socket the chips after extracting them? I'm not sure they would fit in the adapter and it might be too much work soldering on the sockets. Are they easier to resolder them back on if I socket them? Also would they fit nice into the gameshark once I screw back on the plastic front and back?

Also I'm trying to find quickchip on ebay and can't seem to find any. Do you know where I can find some on ebay? I try to do all my online shopping on ebay or paypal if I am to shop online. I also like to avoid products from China as they take a while to ship. If there is something from west coast or midwest United States it would be ideal for fast shipping.

**Comment by parasyte** [comment]
Glad to hear you already have the extractor! Here's how to use it: https://youtu.be/Y5GdqeeQ6oo?t=40s

The chips I linked have 4Mbit capacity, which should be identical to the flash ROMs shipped with GameShark Pro. (I'm really just guessing here, since I'm not at home right now, and don't have any of my notes available here.)

We have tried larger capacity chips, and while the firmware is able to boot with them, it cannot see that there is more capacity available. It also had problems when updating the code list or saving options, because the larger block size erases more than the firmware expects. :) It is effectively hardcoded to only work with 4M x 16 configurations.

Speaking of Kodewerx, I disabled forum registrations a few days ago because spam bots were getting through the filter. I'll be glad to create an account for you, though. Just let me know.

The LPT hardware can be missing on shiny GS Pros, too. There are 2 different hardware revisions that we know of, and one of those revisions just has empty solder pads for the hex inverter and flip-flop. The other revision lacks it completely. Here's the topic you're referring to: http://www.kodewerx.org/forum/viewtopic.php?f=18&t=7189

Yes, that rework station will get the job done. You'll want to use the small nozzle (provides hot air to a more focused area) and keep it moving all around the pins of the chip; not directly to the chip itself. I've always set the temperature to about 300°F (150°C).

Keep in mind that socketing the PCB is *totally optional* but it comes with some benefits:

1. The chips can be easily removed again later if you need to reflash them.
    * This is especially important since it can be easy to accidentally swap the chips before placing them back on the PCB
    * The chips will be held in the sockets by friction instead of solder, so the bond is totally temporary.
2. By soldering the socket, you only have to heat the chips once (to remove them from the board) and maybe a really quick cleanup on each individual pin. This will give you the best confidence that the chips won't be overheated.
    * Without the socket, you'll have to heat them a second time to solder them back on the PCB.
3. In my experience, it's always much easier to desolder than it is to solder. Soldering is a learned skill for sure, but desoldering can be done extremely quickly without focusing on any one single joint. Especially with a hot air gun.
4. The sockets are really easy to solder with the hot air gun; they have "holes" in the bottom for the air to flow freely. It's also pretty easy to use too much heat and melt the plastic a bit, but it's really well temperature tolerant.

And yeah, with low profile sockets, the GS case fits with no problems. It only adds like 1mm of height, overall, and the area surrounding the chips is just air inside the plastic case.

Not sure about solder paste on eBay. I bought a bad batch from a merchant there once, and just had to throw it out. I tend to get specialized electronic components from DigiKey, but I have heard good things about Mouser as well. Both companies keep their stock in the US. That said ... If you search "solder paste" on eBay, you'll get hundreds of results. Just get one that comes in a syringe. I tend to use lead-free low temp solder paste, since I find it easier to work with (and not as damaging to my health!)

**Comment by sharkbyte64** [comment]
Ahh yes I remember that classic public service announcement topic. I first thought the socket would be a big problem if I had to desolder and resolder each time but I think I understand how it works now. It stays on the board forever and I just use the extractor over and over to insert and remove it. Is that correct? My only regret with sockets is you said you had to cut some of the plastic away and I'm afraid I would bend the socket or chip when trying to cut it. Is it required to cut some of the plastic? I burned my hands last time soldering so I will take a few days break and that will happen anyway as it will take a while for all this stuff to ship. I just hope this is enough to fix all the gamesharks I have. What I find weird is that site says "Obsolete item." on the sockets. I found some of the solder paste on ebay. http://www.ebay.com/itm/Chip-Quik-SMD291AX-Solder-Paste-/272528166291?hash=item3f73f1cd93:g:eL4AAOSw4GVYNbPx I think this is the same stuff as the other site you linked. I take it that heat gun I linked would work okay for what I'm working on? One other thing, I noticed on the G540 programmer page it does not show chip support for the 29LE010 chip. Which is weird since that is the programmer you recommended.

**Comment by sharkbyte64** [comment]
I couldn't find a buy it now or add to cart option for that socket. I had to click around a bit and I found this socket. https://www.digikey.com/product-detail/en/mill-max-manufacturing-corp/940-44-032-17-400000/ED90014-ND/947022

Would this work?

**Comment by parasyte** [comment]
You got it! :+1: The socket makes your life much easier.

The low-profile sockets won't fit quite correctly because there's a surface-mount resistor in the way. And it's only a problem because the low-profile sockets have very low clearance from the PCB. I used an X-acto knife to trim away a little bit of the plastic. You could also use a Dremel or similar power tool if you prefer, or even just a sharp knife that you don't mind possibly damaging. I would not recommend trying to solder the socket down unless the pins are perfectly flush with the pads. And that's really only doable when the socket is not touching the resistor.

The obsolete parts warning is useful for companies that need to order large quantities. But for hobbyists, these parts are just fine, and may even be cheaper since they are now just trying to sell off their remaining stock. Of course, if there's none in stock, you'll have to find a suitable replacement!

Now that I'm home, I can check on those part numbers for you. The SST39LF040 that I mentioned before are actually the higher capacity chips. Some comparable chips with the original capacity would be [SST39VF010](http://www.digikey.com/product-detail/en/microchip-technology/SST39VF010-70-4I-NHE/SST39VF010-70-4I-NHE-ND/2297839); these are 1M x 8 and fully compatible. The larger capacity chips just result in a GS Pro that can only use 25% of the available capacity. Which brings me to the next point; you can use any of the following profiles in the G540 software to write the original SST29LE010 chips:

* SST29VF010(PLCC32)
* SST39LF010(PLCC32)
* SST39VF010(PLCC32)

The only differences between these chips and the SST29LE010 are some minor voltage tolerances and the addition of a Sector-Erase command. And (I think) the SST39 uses a newer manufacturing process or something? It's really inconsequential, though.

The links you provided for the rework station, solder paste, and sockets all look good to me. The solder paste has a higher melting point of 361°F (183°C), so you will have to crank up the heat a bit.

**Comment by parasyte** [comment]
And finally, here are some photos of the modification to the sockets. It's a little hard to make out, since the sockets are already mounted. There's just little sliver of the plastic that was removed from the underside of each socket, so they don't touch the resistors. The same mod was done to both sockets, just on different sides.

Bottom socket (the focal point wasn't quite right for this shot, so it's a little blurry):

![img_8445](https://cloud.githubusercontent.com/assets/456942/22723483/a0c90768-ed73-11e6-81b8-cef34accb78b.JPG)

Top socket:
![img_8448](https://cloud.githubusercontent.com/assets/456942/22723486/a4b79c7c-ed73-11e6-999a-ae84a84bdad6.JPG)

**Comment by sharkbyte64** [comment]
Thanks for the pictures. It illustrates it better. I wonder if I could melt some of the plastic away with the soldering iron to create the gap rather than slash away at it with a knife. After I made the post with the solder paste I was going to buy I realized it was not the same as the one you linked and I saw another recommendation at the bottom of the ebay page and I bought this one. http://www.ebay.com/itm/272528166283?_trksid=p2060353.m2749.l2649&ssPageName=STRK%3AMEBIDX%3AIT I also order the heat gun. I haven't bought anything off the site you linked but will probably order tomorrow. Will the new socket link I sent in previous post work? Its a shame that the gamesharks don't use the extra capacity on the newer chips. Heres something that might puzzle you as it puzzled me. Years ago I bought a gameshark pro 3.3 from a local game store. It had 79% codespace free and seemed to have to default codelist of 195 games and almost 2000 codes if not more. Usually the default codelist for 3.3 has only 15% free. When adding or deleting codes the load time would be very slow and take up to 15 seconds. But I still had that codespace free. Until one day I decided to try the piggy back reflash and the codelist glitched down to 15% free. I was really disappointed. Now there is a very weird glitch that goes on with that gameshark and it refuses to add or delete codes and just freezes when you try to do so. Any thoughts on this? Also one of my bricked gamesharks was a result of testing the everdrive 64 v 3 I recently bought. It just reflashed my gameshark which was version 3.1 and I did not think it would do that. What angers me more is that was the only 3.1 I had and I cannot find any online. So this is why I started this project. I really wanted to unbrick that one and put the original 3.1 firmware back on it. I wish they didn't create the everdrive to brick gamesharks instantly. I wish there was a work around to use it. I think I had custom codes I created myself on the 3.1 as well as codes I found on sites years ago that don't exist anymore.

**Comment by parasyte** [comment]
The plastic has pretty good heat resistance, but you can give it a try. And that's the paste I use. Good stuff! The socket is also good.

Corrupted code list, probably. The GameShark is poorly designed, most likely as a cost-saving measure. It's not very safe that the firmware and code list exist on the same rewritable ROMs. And it's also kind of dangerous that both are written in parallel without separating the status reads.

One of the most common ways a GameShark bricks itself is by enabling a "Key Code" (which rewrites the checksums in the ROM header) and you reboot only to find out that it's totally bricked. What we found is that if only one of the two chips acknowledges the write commands, then the checksums will fail to match, and the N64 PIF ROM hangs at boot. This is an example of the warning above that writing in parallel without checking status individually is dangerous.

Speaking of EverDrive 64, it should actually have a decent code engine, now. I wrote it, for crying out loud! It better work ...

**Comment by sharkbyte64** [comment]
I'm not sure I'm using the version of everdrive that has your code engine. All I see is 24 lines of code you have to manually enter and you can't even type in a name of the code or game name to seperate the codes. If there was a way to make the gameshark rom fully working with the everdrive as a preboot to any game you want and figure out a way to have that button that all gamesharks have then it would be flawless as far as gameshark compatibility goes. Also when I was using codes for conkers bad fury day I thought I saved just the moon jump code on the list. The first time moon jump worked. then I played another game, came back and the codes are gone! I would have to manually enter them each time and if i wanted 50 codes it would take forever and I wouldn't be able to tell which is which if I need to turn a few off etc. The other reason I hate the everdrive is it doesn't support saving memory card saves directly to the sd card. If there was someone who coded it to that and the gameshark feature then it would be complete. Maybe they could make an everdrive memory card which fits in the controller, but uses an sd card up to 16GB in size! These were the only two reasons I bought the everdrive. Some promotion site got my hopes up and said it could do those things. Turns out the only thing you can in terms of saving your saves is backing up 1 tiny memory card at a time to a file then putting them on pc. My problem is those memory cards get corrupted like every week and delete all my saves every time.

Speaking of keycodes I got lucky one time just last month and bought some "broken" gamesharks. All I had to do was try over and over different games and eventually 1080 snowboarding did the trick. I wasn't sure which one had the key code or even if it was a key code semi brick but I unbricked all 5 of them with different games. The other bunch of gamesharks I got the month before that I'm afraid are full bricked and need reflashed as I can't get any of them to turn on no matter what game I put in. I have never full bricked a gameshark by putting a keycode in as far as I can remember. Also I take it you can't figure out how that gameshark had 79% codespace free with the default codelist right? I may have a video of that actually if I can find it.

**Comment by sharkbyte64** [comment]
Also is there anything I need to do before desoldering the chips? Do I need to place flux before I desolder with the heat gun, or do I just start it up and move it in a circle around the chip?

**Comment by parasyte** [comment]
Heh, another post with a lot of topics to cover. I'll try to get them all ...

That sounds right with the ED64 menu. The UI is terrible (FWIW the build that I added to [alt64](https://github.com/parasyte/alt64) doesn't even have a UI - at least code management was done with files.) Regardless it should still use the same cheat engine under the hood. So it should support all of the important code types.

AFAIK, it is impossible to get the GameShark firmware working with EverDrive 64 or 64Drive. And in my opinion, it would be a better use of time to build a brand new UI/launcher that can be run from ED64 like any game, or (even better!) replace the ED64 menu entirely, like alt64 does. I wouldn't suggest using alt64 as a base either, because it is poorly coded, completely undocumented, and has the wrong architecture for a GUI application; specifically it is missing an asynchronous event loop. But this is really a tangent... it's something I've been interested in doing myself for quite a while.

The "GS Button" is a different matter. Those codes (ideally) should be converted to use a controller combination to activate the effect.

As far as speculation with your 79% free code list; I assume the code list was just corrupted. There's no way it would be healthy with 2,000 codes and 79% free space. We have seen this issue as well. It's almost always caused by some kind of corruption.

----

For desoldering, just heat it up! The temperature will eventually melt the solder when it hits the melting point, and you can just pluck it off the PCB. I would recommend a suction pen to prevent damage and sliding the pins around molten solder. But you can use tweezers and a steady hand as well. Also check out some video tutorials like this one: https://youtu.be/hw8OGplwefE

Also, keep in mind that these SMD components are soldered in the factory with reflow techniques, which use different temperature profiles for melting the solder evenly. The peak temperature really depends on the composition of the metals in the solder. It's hard to say what they actually used, but to be safe, try lower temperatures first. If it doesn't melt within a minute or two, then you can increase the temp in increments of 50 Degrees (Fahrenheit). 250°F to 300°F is a good place to start.

**Comment by sharkbyte64** [comment]
I would say it is a good corruption if it extends my list to triple the amount they normally hold. I don't know for sure as it is impossible for it to add or remove codes. Its like the firmware is locked. Its all because back a few years ago I used it to try to fix another bricked gameshark and the codelist got jammed. I just love the gameshark 3.3 user interface, the code generator, and hacking on actual gameshark hardware. It brings back so much nostalgia from 1999 and 2000 when my brother and I made simple codes and saved them. Btw, would it be possible for someone with your great expertise and knowledge of programming to create custom firmware for the n64 gameshark? That would be so cool if it could be done. Then we could get version 3.4 or 4.0. I would love to have a gameshark that goes in the actual console under the game and has nearly unlimited codespace with the general codes of every game on the n64 preinstalled. :) Having breakpoints, pointers and other features added to the code generator would be amazing too. Improving the chips so that the firmware and codelist isn't on the same chips would also improve the gameshark so they don't brick as easily. A modernized n64 gameshark would be so cool. I always wondered if someone would homebrew one like that. And after that I always wondered if they could make it so using wifi cartridge we could connect n64s online and play homebrew games online with actual n64 hardware. A lot of this is probably pipe dreaming though. But still would be great if it happened.

**Comment by parasyte** [comment]
I thought about making a replacement firmware, but it's really the same challenge as creating a boot menu for ED64 or 64Drive; the basic utility libraries are available (e.g. libdragon), but a decent event loop is completely lacking. Doing it "the right way" so the code is future proof and portable to other consoles will just take some aggressive architecture and engineering. Something I don't have time to commit to, personally.

On the same subject, I feel it's worthwhile to do such work on ED64 and 64Drive, since they are modern platforms with modern interfaces (USB!) and they don't have the instability or physical characteristics of GameShark.

Have you ever noticed the GameShark is picky about how it sits in the N64? Sometimes it just won't boot unless it's seated *just right*. That's because the male connector pins on the cart edge are a different pitch from every other thing that you plug into your N64. Or have you ever been working on a hack or enjoying some augmented games when your cat rubs against the cartridge tower and the game crashes, taking all of your work and entertainment with it?

We don't have those problems with ED64 or 64Drive, and we really don't want to relive them again. :)

**Comment by sharkbyte64** [comment]
I could not get the usb interface to work at all. All I know is I hate it more than I like it because it killed my gameshark and it doesn't support memory card saves on the sd card. Also I don't see anyway to hack your own codes with the everdrive. I haven't tried alt64 yet either. The only good thing I like is cartridge saves stay permanent (I think) and hopefully don't get corrupted. I have only had 2 games get reset in the 20 years I've had the N64. They only got reset one time so for the most part cartridge saves stay good for a long time. They stay much longer compared to the N64 memory cards. I haven't had the issue of a cat rubbing against the tower and crashing it as I don't have pets now. but even if you were playing any game without the gameshark attached you can't be guaranteed it won't freeze. In my experience the gameshark 3.2 and 3.3 need to be seated slightly higher after you jam it into the N64. Anything 2.2 or lower fits more snug and can be put all the way in. The same thing is true for my copy of Zelda oot 1.1. You move it slightly up or it won't boot at all. But for version 1.0 which I got recently you can have it all the way down. One other thing, since the gameshark has the tab on the bottom completely opened up does that mean it is truely region free? I thought they had action replay in the UK but its basically the same hardware is the gameshark. One other thing, my card got declined from that digi key site. I think I know why though. If it doesn't clear by tomorrow then I don't know why it was declined. They called when I was driving to tell me it was declined and won't tell me why. I stopped and pulled over and told them to check the number again. I wish they supported paypal.

**Comment by parasyte** [comment]
Those are software issues. The hardware is sound, the software needs (a ton of) work, or a complete replacement. You can't hack your own codes with ED64, and that's another software issue.

I think it's "region free" from the hardware perspective (allowing the games to be connected), but the video modes will be troublesome; GS is NTSC only, AR is PAL only, and PAL games don't work well on most NTSC TVs. Actually I haven't tried any PAL video sources on modern TVs, maybe they commonly have PAL decoders, now? Anyway, Japanese games are NTSC, and they'll work great.

**Comment by sharkbyte64** [comment]
I found out only recently ( a couple years ago) that Japanese games have the PAL cut in on the cartridge. They aren't compatible with PAL TVs as far as I know. I have never owned an actual PAL game or japanese game. I guess if I bought a Japanese game I could open the cartridge up with the bit socket security screwdriver and put it in one of my many NTSC cartridges. I was sure it wouldn't work but I tried playing a few PAL games using the everdrive on my NTSC tv and sure as anything they do nothing. I hear sound but no video. Same as my CRT TV except the video can somewhat be seen except the screen moves very quickly in a scrolling pattern. I haven't heard of PAL decoders. Maybe there are some capture cards that will let you see the PAL signal on your computer.

**Comment by parasyte** [comment]
Yes, almost all capture devices support both NTSC and PAL. I bought a [StarTech Composite/USB video adapter](https://www.amazon.com/StarTech-S-Video-Composite-Capture-SVID2USB23/dp/B00535BRBI) a while back, and it's pretty good. The latency is somewhere around 100ms, but still mostly playable.

**Comment by sharkbyte64** [comment]
I tried PAL games with my capture card card and it didn't work. Back when I uploaded gaming videos a lot I used the capture card and used 3 spliters so I could look at the tv while its recording on the computer at the same time. What was weird is I would need to set it at PAL-B for any NES or N64 game to work and the Wii or other modern systems would need NTSC-M. All my consoles are NTSC. My capture card is an EasyCap.

**Comment by parasyte** [comment]
I would love to continue this conversation, but I have to close down this ticket. I think there is enough information here for anyone to fix a bricked GameShark Pro. And my email address was posted earlier for those who are looking for someone willing to do the repair.

If there are similar issues in the future and this ticket doesn't resolve it, don't hesitate to open a new ticket!

**Comment by sharkbyte64** [comment]
k
